src/main/java/site/coach_coach/coach_coach_server/routine/services/RoutineServices.java

```diff
@@ -0,0 +1,70 @@
+package site.coach_coach.coach_coach_server.routine.services;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import site.coach_coach.coach_coach_server.coach.repository.CoachRepository;
+import site.coach_coach.coach_coach_server.matching.Dto.CheckMatchingDto;
+import site.coach_coach.coach_coach_server.matching.repository.MatchingRepository;
+import site.coach_coach.coach_coach_server.routine.domain.Routine;
+import site.coach_coach.coach_coach_server.routine.dto.RoutineForListDto;
+import site.coach_coach.coach_coach_server.routine.dto.RoutineListCoachInfoDto;
+import site.coach_coach.coach_coach_server.routine.dto.RoutineListRequest;
+import site.coach_coach.coach_coach_server.routine.repository.RoutineRepository;
+import site.coach_coach.coach_coach_server.user.domain.User;
+
+@Service
+@RequiredArgsConstructor
+public class RoutineServices {
+	private final RoutineRepository routineRepository;
+	private final MatchingRepository matchingRepository;
+	private final CoachRepository coachRepository;
+
+	public Boolean findIsMatchingServices(RoutineListRequest routineListRequest) {
+		Optional<CheckMatchingDto> checkMatchingDto =
+			matchingRepository.findByUserIdAndCoachId(routineListRequest.userId(),
+				routineListRequest.coachId());
+		if (checkMatchingDto.isEmpty() || checkMatchingDto.get().getIsMatching().equals(false)) {
+			return false;
+		} else {
+			return true;
+		}
```

Comment:
정보공유 2
Optional 에서 map이나 orElse, orElseThrow를 사용할 수도 있습니다

```suggestion
	return matchingRepository.findByUserIdAndCoachId(routineListRequest.userId(), routineListRequest.coachId())
            .map(CheckMatchingDto::getIsMatching)
            .orElse(false);
```

Reply:
오 깔끔해지고 좋네요! 한번 적용해 보겠습니다!